airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/avro/JsonToAvroSchemaConverter.java

```diff
@@ -240,6 +254,11 @@ Schema getNullableFieldTypes(final String fieldName, final JsonNode fieldDefinit
       if (!nonNullFieldTypes.contains(NULL_SCHEMA)) {
         nonNullFieldTypes.add(0, NULL_SCHEMA);
       }
+      if ((nonNullFieldTypes
+          .stream().anyMatch(schema -> schema.getLogicalType() != null)) &&
+          (!nonNullFieldTypes.contains(Schema.create(Schema.Type.STRING)))) {
```

Comment:
It's probably better to save `Schema.create(Schema.Type.STRING))` as a static private variable to avoid creating the string schema repeatedly. This `if` block does it twice: one in the check, and one to be appended to the field type list.

Reply:
done